packages/apps/reputation-oracle/server/src/modules/reputation/reputation.dto.ts

```diff
@@ -53,6 +57,34 @@ export class ReputationGetAllQueryDto {
   @IsOptional()
   @Transform(({ value }) => (Array.isArray(value) ? value : [value]))
   public roles?: ReputationEntityType[];
+
+  @ApiPropertyOptional({
+    enum: ReputationOrderBy,
+    default: ReputationOrderBy.CREATED_AT,
+  })
+  @IsEnum(ReputationOrderBy)
+  @IsIn(Object.values(ReputationOrderBy))
+  @IsOptional()
+  public orderBy?: ReputationOrderBy;
+
+  @ApiPropertyOptional({
+    enum: SortDirection,
+    default: SortDirection.DESC,
+  })
+  @IsEnum(SortDirection)
+  @IsIn(Object.values(SortDirection))
+  @IsOptional()
+  public orderDirection?: SortDirection;
+
+  @ApiPropertyOptional({ type: Number })
+  @IsOptional()
+  @Transform(({ value }) => Number(value))
```

Comment:
Should `first` and `skip` have `Min(N)` for validation purpose?

Reply:
Yes, agree

Reply:
```
@ApiPropertyOptional({ type: Number })
  @IsOptional()
  @Min(1)
  @Transform(({ value }) => Number(value))
  public first?: number;

  @ApiPropertyOptional({ type: Number })
  @IsOptional()
  @Min(0)
  @Transform(({ value }) => Number(value))
  public skip?: number;
```

Reply:
What do you think? @dnechay